src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -1046,6 +1028,9 @@ public Task<bool> HasSuccessfullyLoadedAsync(ProjectState project, CancellationT
                 return compilation.ToMetadataReference(projectReference.Aliases, projectReference.EmbedInteropTypes);
             }
```

Comment:
this now always succeed when it's from teh same language.  the call to tracker.GetCompilationAsync above will ensure that you get back a compilation with all the references properly added (modulo respecting the 'generate skeletons' flag).